### DISCUSSION TITLE
support for colored output?

### DISCUSSION THREAD

**Discussion by lukehsiao** [pacak/bpaf#77] [Q&A] [answered]
One nice thing about clap is the colored output. Is this something `bpaf` might support in the future?

**Comment by pacak** [comment, marked as answer]
Help rendering code is decoupled from parsing so it should be easy to add but I don't know if people are interested in it or not. Let's keep it open and once there's some more votes - I'll implement something.

**Reply by pacak** [reply]
Version 0.7.0 supports two types of colored output with cargo features `bright-color` and `dull-color` for roughly clap 3 and clap 4 color schemes.

**Reply by lukehsiao** [reply]
Well, that's amazing! Thanks!

**Reply by pacak** [reply]
Let me know if it works as you would expect, maybe something can be improved - I prefer boring b/w so I don't know :)

**Reply by dantheman3333** [reply]
hi @pacak I just tried it out, thanks for adding this!
I have some suggestions:

- `dull-color` is making words bold and slightly brighter, but still white, so maybe it should be called color-white-highlight or something similar? I wasn't sure if it was working at first 
- `bright-color` produces yellow and green colors that are more dull than cargo's, maybe they could match? 

(tried using ubuntu's standard terminal)

what are your thoughts?

**Reply by pacak** [reply]
Interesting. They match for me... Can you post a screenshot comparing those two? `dull-color` should be similar to what `clap` 4 produces and `bright-color` matches `clap 3... At least visually in my terminal, I haven't looked at exact code it generates, maybe I shoud.

**Reply by dantheman3333** [reply]
interesting, I added some screenshots here with an attempt to fix without diving further, but I guess I should have https://github.com/pacak/bpaf/pull/106 
edit: might just be the `bold` that makes the difference

**Reply by pacak** [reply]
I see. I'm using somewhat exotic terminal and they look the same to me :) Thank you for your contribution, I'll get it merged shortly.
